### PR TITLE
feat(sink): add es retry_on_conflict and max_task_num

### PR DESCRIPTION
**Pull request by xxhZs**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

max_task_num：Maximum number of lines written to es at the same time
retry_on_conflict：Number of retries for es ooc

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

We add two optional option for es sink
`retry_on_conflict `: default is 3. The num of retries after an optimistic lock failure for es
`batch_size_kb  `: the max size of each request batch
`batch_num_messages`: the max rows of each request batch
（batch exceeds the limit of `batch_size_kb   `or `batch_num_messages`, we will send a batch）
`concurrent_requests `:the max number of threads per degree of parallelism

</details>

### CODE REVIEW

**Comment by zwang28** [comment]
Is it possible these properties be made configurable, e.g. from the toml, rather than being fixed during sink creation?

File: java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java [hunk 1]
```diff
@@ -68,12 +74,17 @@ void addOkResult(int numberOfActions) {
...
             taskCount++;
             EsWriteResultResp esWriteResultResp;
             while (true) {
-                if ((esWriteResultResp = this.blockingQueue.poll()) != null) {
+                if ((esWriteResultResp = this.blockingQueue.poll(10, TimeUnit.MILLISECONDS))
+                        != null) {
                     checkEsWriteResultResp(esWriteResultResp);
+                }
+
+                if (taskCount >= maxTaskNum) {
```
Comment (hzxa21) [inline java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java:86]: ES bulkprocessoer seems to be built-in support for batching and concurrency limit. Should we use `setBulkActions(maxTaskNum)` instead?

Comment (xxhZs) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java:86]: It looks like setBulkActions is used to set the batch size, and this is to set the parallelism of writing to es. They seem to control two things

Comment (hzxa21) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java:86]: IIUC, `setConcurrentRequests` is used to control the concurrency and `setBulkActions` is used to control the batch size. We use `setConcurrentRequests(1)` in es sink so I guess the concurrency is always 1. That is why I thought the `maxTaskNum` is used to control batch size.

Comment (xxhZs) [reply java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSink.java]: fix

File: java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/ElasticBulkProcessorAdapter.java [hunk 2]
```diff
@@ -47,19 +50,20 @@ public ElasticBulkProcessorAdapter(
                         new BulkListener(requestTracker));
         // Possible feature: move these to config
         // execute the bulk every 10 000 requests
```
Comment (hzxa21) [inline java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/ElasticBulkProcessorAdapter.java:52]: nit: fix the doc

File: java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSinkConfig.java [hunk 3]
```diff
@@ -41,6 +41,18 @@ public class EsSinkConfig extends CommonSinkConfig {
     @JsonProperty(value = "index_column")
     private String indexColumn;
 
+    @JsonProperty(value = "retry_on_conflict")
+    private Integer retryOnConflict;
+
+    @JsonProperty(value = "bulk_actions")
+    private Integer bulkActions;
+
+    @JsonProperty(value = "bulk_size")
```
Comment (hzxa21) [inline java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSinkConfig.java:50]: `batch_size_kb`

File: java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSinkConfig.java [hunk 4]
```diff
@@ -41,6 +41,18 @@ public class EsSinkConfig extends CommonSinkConfig {
     @JsonProperty(value = "index_column")
     private String indexColumn;
 
+    @JsonProperty(value = "retry_on_conflict")
+    private Integer retryOnConflict;
+
+    @JsonProperty(value = "bulk_actions")
```
Comment (hzxa21) [inline java/connector-node/risingwave-sink-es-7/src/main/java/com/risingwave/connector/EsSinkConfig.java:47]: `batch_num_messages`

**Review by hzxa21 — APPROVED** [review]
Rest LGTM

**Pull request merged**